graphql_compiler/schema_generation/orientdb/get_orientdb_schema_graph.py

```diff
@@ -0,0 +1,488 @@
+    vertex_class_names = _get_class_names_of_kind(inheritance_sets, Kind.Vertex)
+    non_graph_class_names = _get_class_names_of_kind(inheritance_sets, Kind.NonGraph)
+
+    for class_name in edge_class_names:
+        class_definition = class_name_to_definition[class_name]
+        class_fields = _get_class_fields(class_definition)
+        abstract = _is_abstract(class_definition)
+
+        inherited_property_definitions = _get_inherited_property_definitions(
+            inheritance_sets[class_name], class_name_to_definition)
+        link_property_definitions, non_link_property_definitions = (
+            _get_link_and_non_link_properties(inherited_property_definitions))
+
+        for definition in link_property_definitions:
+            _validate_link_definition(
+                class_name_to_definition, definition, vertex_class_names, subclass_sets)
+
+        links = _get_end_direction_to_superclasses(link_property_definitions)
+
+        maybe_base_in_connection, maybe_base_out_connection = _try_get_base_connections(
+            class_name, inheritance_sets, links, abstract)
+
+        property_name_to_descriptor = _get_element_properties(
+            class_name, non_link_property_definitions, non_graph_class_names, inner_collection_objs)
+
+        edge_elements[class_name] = EdgeType(
+            class_name, abstract, property_name_to_descriptor, class_fields,
+            maybe_base_in_connection, maybe_base_out_connection)
+    return edge_elements
+
+
+def _get_vertex_elements(class_name_to_definition, inheritance_sets, inner_collection_objs):
+    """Return a dict mapping class name to VertexType."""
+    vertex_elements = dict()
+
+    vertex_class_names = _get_class_names_of_kind(inheritance_sets, Kind.Vertex)
+    non_graph_class_names = _get_class_names_of_kind(inheritance_sets, Kind.NonGraph)
+
+    for class_name in vertex_class_names:
+        class_definition = class_name_to_definition[class_name]
+        class_fields = _get_class_fields(class_definition)
+        abstract = _is_abstract(class_definition)
+
+        inherited_property_definitions = _get_inherited_property_definitions(
+            inheritance_sets[class_name], class_name_to_definition)
+        link_property_definitions, non_link_property_definitions = (
+            _get_link_and_non_link_properties(inherited_property_definitions))
+
+        if len(link_property_definitions) > 0:
+            raise AssertionError(u'There are links {} defined on non-edge class {}'
+                                 .format(link_property_definitions, class_name))
+
+        property_name_to_descriptor = _get_element_properties(
+            class_name, non_link_property_definitions, non_graph_class_names, inner_collection_objs)
+
+        vertex_elements[class_name] = VertexType(
+            class_name, abstract, property_name_to_descriptor, class_fields)
+    return vertex_elements
+
+
+def _get_class_names_of_kind(inheritance_sets, kind):
```

Comment:
It's weird that `_get_class_names_of_kind` would need `inheritance_sets`. If you renamed the function to `get_subclasses(inheritance_sets, class_name)`, it will be clear with no extra documentation why the `inheritance_sets` are needed. This will also get rid of all use cases of the `Kind` enum, which sounds like it has to do with metaprogramming.

Reply:
This is not quite true.  Finding the non-graph class names would involve finding the set difference between all the class names and union of the vertex and edge class names. I am going to reuse what we had before: `_split_elements_of_kind`. However, I am going to modify it to use the subclass_sets instead. I think it makes more sense.

Reply:
I removed _get_class_names_of_kind. At the end, I decided against using subclass_sets instead of inheritance_sets. I would had to compute the subclass_sets in several parts of the code and that would make this pr even larger. I kept this code as much as a copy paste as possible by just modifying `_split_classes_of_kind` into the function `_get_vertex_edge_and_non_graph_class_names`.